presto-main/src/main/java/com/facebook/presto/util/DateTimeUtils.java

```diff
@@ -548,4 +552,52 @@ else if (parsePos < 0) {
             return ~bestInvalidPos;
         }
     }
+
+    public static SqlTimestamp sqlTimestampOf(int year,
+            int monthOfYear,
+            int dayOfMonth,
+            int hourOfDay,
+            int minuteOfHour,
+            int secondOfMinute,
+            int millisOfSecond,
+            DateTimeZone baseZone,
+            TimeZoneKey timestampZone,
```

Comment:
eg. In old semantic when literal WITH TZ is casted to TIMESTAMP W/O TZ and TZ of literal is different than session TZ.